Birthday-2024-Project/Scripts/GameManager.gd

```diff
@@ -154,30 +164,35 @@ func _rotate_held_piece():
 		held_piece.rotate_clockwise()
 		held_piece.play_rotate_audio()
 		_reset_settled()
+		remaining_settle_delay = held_piece.ROTATION_ANIMATION_DURATION
 	elif Input.is_action_just_pressed("RotateAnticlockwise"):
 		held_piece.rotate_anticlockwise()
 		held_piece.play_rotate_audio()
 		_reset_settled()
-
+		remaining_settle_delay = held_piece.ROTATION_ANIMATION_DURATION
 
 func _get_held_piece_grid_origin() -> Vector2i:
 	return grid.PositionToGridCoordinate(held_piece.GetOriginCellPosition())
 
 func _do_held_piece_settle(delta):
 	var mouse_position : Vector2 = get_global_mouse_position()
 	var mouse_distance_moved : float = (mouse_position - previous_mouse_position).length()
-	if mouse_distance_moved > 0.1:
+	var mouse_grid_pos : Vector2i = grid.PositionToGridCoordinate(mouse_position)
+	if (mouse_distance_moved > 0.1 and mouse_grid_pos != previous_mouse_grid_pos) or grid.CheckIfOffGridPos(mouse_position):
```

Comment:
This specific magic number I think shouldn't be a constant, since as far as I understand it's just a "did the mouse actually move beyond possible floating point errors?" number with no significance. I tried to make it more readable by putting the logic in a separate function with a comment